### PR TITLE
feat(orderBy): Stable sort the input array

### PR DESCRIPTION
**Pull request by lgalfaso**
Stable sort the input array

### CODE REVIEW

**Comment by lgalfaso** [comment]
This indirectly fixes #12405

**Comment by Narretz** [comment]
Does this have a performance implications?

**Comment by lgalfaso** [comment]
> Does this have a performance implications?
>
 This only adds a natural comparator with the original element position, so
this adds some cycles but much smaller than even one $parse evaluation

File: src/ng/filter/orderBy.js [hunk 1]
```diff
@@ -183,6 +183,7 @@ function orderByFilter($parse) {
     if (sortPredicate.length === 0) { sortPredicate = ['+']; }
 
     var predicates = processPredicates(sortPredicate, reverseOrder);
+    predicates.push({ get: function() { return {}; }, descending: reverseOrder ? -1 : 1});
```
Comment (caitp) [inline src/ng/filter/orderBy.js]: I think we should leave a comment explaining this, it won't be very nice to have to look up this issue in the blame for this function to figure out what this means once it stops being clear.

Just a quick header comment with an overview explaining how the predicate works, imho

**Comment by caitp** [comment]
> This only adds a natural comparator with the original element position, so
this adds some cycles but much smaller than even one $parse evaluation

It might still be a good idea to look at the impact this has on the bigtable benchmarks, and put some numbers in the commit message and issue --- so that there's a bit more confidence there. Can we do that?

**Comment by lgalfaso** [comment]
Run order_by benchpress several times with and without the patch

With the patch
mean: 152.74ms ± 22% 
(stddev 33.61)

Without the patch
mean: 156.36ms ± 32% 
(stddev 53.76) 


This PR does not make the code any faster (as the numbers look like they claim), but do show that the variations between runs is much bigger than whatever performance penalty this patch adds

**Comment by Narretz** [comment]
Can you add to the commit message that it fixes #12405? Otherwise LGTM

**Pull request closed without merging**